### PR TITLE
Don't block forever in find-new-repl-db / handle out-of-order physreps gracefully

### PR DESCRIPTION
**Pull request by markhannum**
This PR also introduces 'incoherent_test' under phys_rep_tiered which attempts to reproduce these situations.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>11/579 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>misstable_remsql [setup failure]<br>basic_ssl_generated [setup failure]<br>sc_resume_logicalsc_generated<br>analyze_exit_immediately<br>timepart_trunc_serialsc_generated<br>timepart_trunc<br>snap_ha_retry_newsi_generated<br>tunables<br>truncatesc_offline_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>5/579 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sidupidx3 [setup failure]<br>writes_remsql_names_negativerowid_generated [setup failure]<br>sc_resume_logicalsc_generated<br>commit_lsn_map_delete_generated<br>snap_ha_retry_newsi_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>9/579 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>comdb2sys [setup failure]<br>writes_remsql_names_rte_connect_generated [setup failure]<br>writes_remsql_names_negativerowid_generated [setup failure]<br>basic_rowset_generated [setup failure]<br>sc_resume_logicalsc_generated<br>sc_resume<br>online_compaction<br>snap_ha_retry_newsi_generated<br>sc_truncate_lockorder_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>8/579 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_newuniq_logicalsc_generated [setup failure]<br>writes_remsql_names_negativerowid_generated [setup failure]<br>sc_remsql [setup failure]<br>basic_nogenid48_generated [setup failure]<br>sc_resume_logicalsc_generated<br>sc_resume<br>snap_ha_retry_newsi_generated<br>dbcreate</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>6/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_newuniq_logicalsc_generated [setup failure]<br>sc_resume<br>queuedb_rollover_noroll1_generated<br>timepart_trunc<br>snap_ha_retry_newsi_generated<br>auth</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>277/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_inserts_logicalsc_generated [setup failure]<br>scupdates_logicalsc_generated [setup failure]<br>sc_resume<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated<br>comdb2sys<br>timepart_trunc<br>auth_twofiles_generated<br>auth_queueodh_generated<br>auth</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>249/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_newuniq_logicalsc_generated [setup failure]<br>sc_redo<br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated<br>comdb2sys<br>auth_twofiles_generated<br>auth_queueodh_generated<br>auth<br>userschema_queueodh_generated<br>userschema</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>222/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>comdb2sys_queueodh_generated<br>comdb2sys_pagesize_generated<br>comdb2sys<br>commit_lsn_map_delete_generated<br>auth_twofiles_generated<br>auth_queueodh_generated<br>auth<br>userschema_queueodh_generated<br>userschema<br>snap_ha_retry_newsi_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>9/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>writes_remsql_names_fdbpushredirect_generated [setup failure]<br>basic_rowlocks_generated [setup failure]<br>basic_protobuf_connectmsg_generated [setup failure]<br>verify_db_parallel_generated [setup failure]<br>sc_resume_logicalsc_generated<br>sc_transactional_rowlocks_generated<br>commit_lsn_map_delete_generated<br>snap_ha_retry_newsi_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>10/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>simple_timepart_reptimeout_generated [setup failure]<br>comdb2sys [setup failure]<br>selectv_range [setup failure]<br>load_cache [setup failure]<br>ssl_remsql_rte_connect_generated [setup failure]<br>blob_sanity [setup failure]<br>analyze<br>commit_lsn_map_delete_generated<br>snap_ha_retry_newsi_generated</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/580 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_lotsoftables_logicalsc_generated [setup failure]<br>sc_inserts_logicalsc_generated [setup failure]<br>writes_remsql_names_fdbpush_generated [setup failure]<br>snap_ha_retry_newsi_generated</p>

File: lua/lib/physrep_register_replicant.lua [hunk 1]
```diff
@@ -5,6 +5,12 @@
 -- register themselves. The node, in return, sends back a list of "potential"
 -- nodes that the replicant can connect to to pull and apply physical logs.
 local function main(dbname, hostname, lsn, source_dbname, source_hosts)
+
+    print(
```
Comment (aakash10292) [inline lua/lib/physrep_register_replicant.lua:9]: nitpick : maybe useful to have physrep_fanout and physrep_max_candidates as well here ?

**Review by aakash10292 — APPROVED** [review]
LGTM

**Pull request merged**